Nb cosmicslime

stephjuneau opened the pull request:
New notebook by @i-olin to showcase the Cosmic Slime MCPM Value-Added Catalog in SDSS DR17. Also updated the keywords list to add one more keyword. The goal is to have a first reasonable version of this notebook before the AAS meeting and we can further improve or streamline it later, if needed.

I just completed my own review (leading to this revised version) but let me know if someone else is available to take a look or merge the PR (@rnikutta @jacquesalice @bmerino95 @kareninysimba etc.).

bmerino95 commented:
Hi @stephjuneau and @i-olin! I have looked over the notebook, and overall, it looks great! Here is my feedback:

- I only found one spelling mistake. The word' simultaneously' is misspelled in the first code block under Query Cosmic Slime Catalog.

- I found an issue in Plot Query Results:
The last three lines of the first code block say that I can uncomment a line to locate the Coma Cluster. 
When I uncommented the line and ran the cell, I received an error message pointing to line 18 that said, `“AttributeError: 'BarContainer' object has no attribute 'Rectangle'”`
You can replace the current line with:
`ax.add_patch(Rectangle((190, 25),10,7,fill=False,color='white'))`
To get the desired box identifying the Coma cluster. 

- In Cosmic Slime Galaxy Redshift Distribution:
Royalblue is included in colors, but instead of using the list, the color is set to royalblue manually for run0.
Would it be worth replacing lines 22 and 23 with the following?
`ax.hist(zbin,bins=Nbins,histtype='stepfilled',color=colors[i],alpha=0.1)`
`ax.hist(zbin,bins=Nbins,histtype='step',color=colors[i],label='Run 0',linewidth=3)`
Instead of:
`ax.hist(zbin,bins=Nbins,histtype='stepfilled',color='royalblue',alpha=0.1)`
`ax.hist(zbin,bins=Nbins,histtype='step',color='royalblue',label='Run 0',linewidth=3)`

- I have a question about some of the references:
The following references are used, but I think the years might be incorrect. 

1. Rines et al. 2003

ApJ lists the citation as:
Citation Kenneth Rines et al 2005 AJ 130 1482
So, should the citation be Rines et al. 2005?

2. Tiwari et al. 2005

This one is a bit confusing. The BibTex citation says Tiwari et al. 2020, but Arxiv’s ‘Cite as’ option uses 2005. 

3. Elek et. al 2020

The first minor thing is that there is a period after ‘et’ in this citation in references.
Second, The MIT Press Direct citation for this article is:
Oskar Elek, Joseph N. Burchett, J. Xavier Prochaska, Angus G. Forbes; Monte Carlo Physarum Machine: Characteristics of Pattern Formation in Continuous Stochastic Transport Networks. Artif Life 2022; 28 (1): 22–57. doi: https://doi.org/10.1162/artl_a_00351
So should the citation be Elek et al. 2022?

stephjuneau commented:
> Hi @stephjuneau and @i-olin! I have looked over the notebook, and overall, it looks great! Here is my feedback:
> 
Thanks for the detailed review, @bmerino95, much appreciated! I'll tag you as official reviewer and when you check the fixes, someone can merge the PR.

> * I only found one spelling mistake. The word' simultaneously' is misspelled in the first code block under Query Cosmic Slime Catalog.

Fixed

> * I found an issue in Plot Query Results:
>   The last three lines of the first code block say that I can uncomment a line to locate the Coma Cluster.
>   When I uncommented the line and ran the cell, I received an error message pointing to line 18 that said, `“AttributeError: 'BarContainer' object has no attribute 'Rectangle'”`
>   You can replace the current line with:
>   `ax.add_patch(Rectangle((190, 25),10,7,fill=False,color='white'))`
>   To get the desired box identifying the Coma cluster.

Thanks for catching that! Fixed.

> * In Cosmic Slime Galaxy Redshift Distribution:
>   Royalblue is included in colors, but instead of using the list, the color is set to royalblue manually for run0.
>   Would it be worth replacing lines 22 and 23 with the following?
>   `ax.hist(zbin,bins=Nbins,histtype='stepfilled',color=colors[i],alpha=0.1)`
>   `ax.hist(zbin,bins=Nbins,histtype='step',color=colors[i],label='Run 0',linewidth=3)`
>   Instead of:
>   `ax.hist(zbin,bins=Nbins,histtype='stepfilled',color='royalblue',alpha=0.1)`
>   `ax.hist(zbin,bins=Nbins,histtype='step',color='royalblue',label='Run 0',linewidth=3)`

Sure, that works!

> * I have a question about some of the references:
>   The following references are used, but I think the years might be incorrect.
> 
> 1. Rines et al. 2003
> 
> ApJ lists the citation as: Citation Kenneth Rines et al 2005 AJ 130 1482 So, should the citation be Rines et al. 2005?
> 
> 2. Tiwari et al. 2005
> 
> This one is a bit confusing. The BibTex citation says Tiwari et al. 2020, but Arxiv’s ‘Cite as’ option uses 2005.
> 
> 3. Elek et. al 2020
> 
> The first minor thing is that there is a period after ‘et’ in this citation in references. Second, The MIT Press Direct citation for this article is: Oskar Elek, Joseph N. Burchett, J. Xavier Prochaska, Angus G. Forbes; Monte Carlo Physarum Machine: Characteristics of Pattern Formation in Continuous Stochastic Transport Networks. Artif Life 2022; 28 (1): 22–57. doi: https://doi.org/10.1162/artl_a_00351 So should the citation be Elek et al. 2022?

Fixed all those dates (thanks again!) and also added the journal info.

bmerino95 reviewed (approved):
Verified the changes. Everything looks good!

stephjuneau commented:
Thanks, @bmerino95 !

rnikutta commented:
Hi @stephjuneau and @i-olin very nice NB! Just a few more suggesions:

- Add Stephanie to the `__author__` in the first cell.
- Add the next free `__nbid__` (check the nbid.csv) file. And update the nbid.csv file too.
- Why is "Matter Density" capitalized? (twice in Goals, and also elsewhere in the NB)
- I'd remove "Please note that this notebook is written for Python 3." in Imports; they all are by now.
- Imports cell: move SPARCL imports below "3rd party". For consistency, we usually go: std lib, 3rd party, DL stuff..
- Remove the getpass import line, it's not used in the NB.
- Comment in a cell: "# Call query client and save output as Astropy Table" --> Better "return an Astropy Table" (since it doesn't save anything.) This is in multiple cells.
- Typo: "The Cosmic Slime Value Added catalog was released with SDSS DR17 and can found here," --> "and can be found here"
- You can avoid the `<matplotlib.legend.Legend at 0x7f5cf9185e40>` lines being displayed just above a figure, if you end the last plotting instruction of the cell end with `;`, e.g. `plt.legend();`
- Some references could use an extra space, e.g. "Rines et al.2003" --> "Rines et al. 2003" (and others)
- PEP8 recommends: variables listed in a multi-assigment get a space after a comma, e.g. `dens_1, _, _`
- Caption right panel of figure on Matter Density Contours: it says "MatterDens". Should be "matter density"?

stephjuneau commented:
Hi @rnikutta , thanks for the additional careful review! All your suggestions have been implemented and we also fixed the year for two references in the main text (which Brian had noted for the References section: Rines et al. 2005 rather than 2003 and Tiwari et al. 2020 rather than 2005). I also added the `nbid` so please see if it now looks ready to be merged?

> Hi @stephjuneau and @i-olin very nice NB! Just a few more suggesions:
> 
> * Add Stephanie to the `__author__` in the first cell.
> * Add the next free `__nbid__` (check the nbid.csv) file. And update the nbid.csv file too.
> * Why is "Matter Density" capitalized? (twice in Goals, and also elsewhere in the NB)
> * I'd remove "Please note that this notebook is written for Python 3." in Imports; they all are by now.
> * Imports cell: move SPARCL imports below "3rd party". For consistency, we usually go: std lib, 3rd party, DL stuff..
> * Remove the getpass import line, it's not used in the NB.
> * Comment in a cell: "# Call query client and save output as Astropy Table" --> Better "return an Astropy Table" (since it doesn't save anything.) This is in multiple cells.
> * Typo: "The Cosmic Slime Value Added catalog was released with SDSS DR17 and can found here," --> "and can be found here"
> * You can avoid the `<matplotlib.legend.Legend at 0x7f5cf9185e40>` lines being displayed just above a figure, if you end the last plotting instruction of the cell end with `;`, e.g. `plt.legend();`
> * Some references could use an extra space, e.g. "Rines et al.2003" --> "Rines et al. 2003" (and others)
> * PEP8 recommends: variables listed in a multi-assigment get a space after a comma, e.g. `dens_1, _, _`
> * Caption right panel of figure on Matter Density Contours: it says "MatterDens". Should be "matter density"?

rnikutta commented:
Thank you @stephjuneau , merging.
